src/torchcodec/decoders/_core/VideoDecoder.cpp

```diff
@@ -1342,37 +1349,93 @@ void VideoDecoder::convertAudioAVFrameToFrameOutputOnCPU(
       !preAllocatedOutputTensor.has_value(),
       "pre-allocated audio tensor not supported yet.");
 
-  const AVFrame* avFrame = avFrameStream.avFrame.get();
+  AVSampleFormat sourceSampleFormat =
+      static_cast<AVSampleFormat>(avFrameStream.avFrame->format);
+  AVSampleFormat desiredSampleFormat = AV_SAMPLE_FMT_FLTP;
+
+  UniqueAVFrame convertedAVFrame;
+  if (sourceSampleFormat != desiredSampleFormat) {
+    convertedAVFrame = convertAudioAVFrameSampleFormat(
+        avFrameStream.avFrame, sourceSampleFormat, desiredSampleFormat);
+  }
+  const UniqueAVFrame& avFrame = (sourceSampleFormat != desiredSampleFormat)
+      ? convertedAVFrame
+      : avFrameStream.avFrame;
```

Comment:
Above: I wasn't able to use the call to `convertAudioAVFrameSampleFormat` within the ternary expression.

Reply:
Actually, I just realized that `convertAudioAVFrameSampleFormat()` already has that exact signature. I think we could simplify things a lot if `convertAudioAVFrameSampleFormat()` became `maybe ConvertAudioAVFrameSampleFormat()` (or a better name?) and only did the conversion if needed.

Reply:
I wasn't able to make it work.

The function cannot take a `UniqueAVFrame avFrame` as input, it has to be a ref (or const ref).

Then it has to return either `avFrame` (which is a ref) or `convertedAVFrame` which is a `UniqueAVFrame` allocated within that function.

I tried a few things, but nothing worked. It's possible I'm doing something wrong.

Reply:
I think there may be a way to do it, but it requires doing some `std::move()` calls. I've had to do that on both sides: going in and out of functions. Let's commit this code as-is, and I can try playing with it later.